mllib/src/main/scala/org/apache/spark/mllib/feature/StandardScaler.scala

```diff
@@ -61,20 +61,30 @@ class StandardScaler(withMean: Boolean, withStd: Boolean) extends Logging {
  * :: Experimental ::
  * Represents a StandardScaler model that can transform vectors.
  *
- * @param withMean whether to center the data before scaling
- * @param withStd whether to scale the data to have unit standard deviation
  * @param mean column mean values
  * @param variance column variance values
+ * @param withMean whether to center the data before scaling
+ * @param withStd whether to scale the data to have unit standard deviation
  */
 @Experimental
-class StandardScalerModel private[mllib] (
-    val withMean: Boolean,
-    val withStd: Boolean,
+class StandardScalerModel (
     val mean: Vector,
-    val variance: Vector) extends VectorTransformer {
+    val variance: Vector,
+    private var withMean: Boolean = false,
+    private var withStd: Boolean = true) extends VectorTransformer {
 
```

Comment:
The default argument is not friendly for Java though; why don't we add another constructor which takes only mean and variance?